manifests/mirror.pp

```diff
@@ -86,6 +98,18 @@
     $components_arg = " ${components}"
   }
 
+  if empty($filter) {
+    $filter_arg = ''
+  } else{
+    $filter_arg = "-filter=\"${filter}\""
+  }
+
+  if ($filter_with_deps == true) {
+    $filter_with_deps_arg = '-filter-with-deps'
```

Comment:
If you add a space before and after the value being assigned to `$filter_arg` then the spec tests for when `filter_arg` is not present wouldn't need to rely on consecutive blank whitespace. Which would also make the change set a fair bit smaller.